applications/GeoMechanicsApplication/custom_utilities/solving_strategy_factory.hpp

```diff
@@ -44,6 +44,11 @@ class SolvingStrategyFactory
         if (rSolverSettings[strategy_type].GetString() ==
             GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::Name())
         {
+            const int max_iterations = rSolverSettings["max_iterations"].GetInt();
```

Comment:
@avdg81 Do you think we need to check this with 'Has' or is it fine if it throws when it doesn't have these parameters

Reply:
Since it seems that these settings are optional rather than required I agree with you that checking whether any particular setting exists is probably the better way to go. However, you could have any combination of optional settings (some may be present, others are not), which complicates things. In that case, we would need to know about the default values for the undefined settings, which is not straightforward. So, for now, I would suggest to keep it the way it is.